python/src/spark_rapids_ml/core.py

```diff
@@ -313,7 +405,7 @@ def _pyspark_class(self) -> Optional[ABCMeta]:
         )
 
 
-class _CumlCaller(_CumlParams, _CumlCommon):
+class _CumlCaller(HasEnableSparseDataOptim, _CumlParams, _CumlCommon):
```

Comment:
This suggests we support sparse vectors in all our algos, while only logistic regression has support.   Seems it will break if sparse vectors are fed into other algos.

Reply:
Right. Moved the param to the algorithm class LogisticRegression.